app/graphql/types/base_scalar.rb

```diff
@@ -0,0 +1,4 @@
+module Types
+  class BaseScalar < GraphQL::Schema::Scalar
+  end
+end
```

Comment:
Actually, we can remove:

```
app/graphql/types/base_argument.rb
app/graphql/types/base_enum.rb
app/graphql/types/base_field.rb
app/graphql/types/base_input_object.rb
app/graphql/types/base_interface.rb
app/graphql/types/base_union.rb
```

This is all boilerplate code and since we added no custom logic there, we don't need these files.
I would keep `BaseMutation` though, in case we later need to add custom Mutation logic (kind of like having an `ApplicationController` instead of inheriting from `ActionController::Base`.

Reply:
Done in https://github.com/consul/consul/pull/4766/commits/59ce9a85bce016a8faa845fd890a5f2fb40caabf